src/deephaven_ib/__init__.py

```diff
@@ -0,0 +1,894 @@
+        # Market Data
+        ####
+
+        ticks_price: real-time tick market data of price values requested via 'request_market_data'.
+        ticks_size: real-time tick market data of size values requested via 'request_market_data'.
+        ticks_string: real-time tick market data of string values requested via 'request_market_data'.
+        ticks_efp: real-time tick market data of exchange for physical (EFP) values requested via 'request_market_data'.
+        ticks_generic: real-time tick market data of generic values requested via 'request_market_data'.
+        ticks_option_computation: real-time tick market data of option computations requested via 'request_market_data'.
+        ticks_trade: real-time tick market data of trade prices requested via 'request_tick_data_historical' or 'request_tick_data_realtime'.
+        ticks_bid_ask: real-time tick market data of bid and ask prices requested via 'request_tick_data_historical' or 'request_tick_data_realtime'.
+        ticks_mid_point: real-time tick market data of mid-point prices requested via 'request_tick_data_historical' or 'request_tick_data_realtime'.
+        bars_historical: historical price bars requested via 'request_bars_historical'
+        bars_realtime: real-time price bars requested via 'request_bars_realtime'
+
+        ####
+        # Order Management System (OMS)
+        ####
+
+        orders_open: open orders.  Automatically populated.
+        orders_status: order statuses.  Automatically populated.
+        orders_completed: completed orders.  Automatically populated.
+        orders_exec_details: order execution details.  Automatically populated.
+        orders_exec_commission_report: order execution commission report.  Automatically populated.
+    """
+
+    _client: IbTwsClient
+
+    def __init__(self, download_short_rates=True):
+        self._client = IbTwsClient(download_short_rates=download_short_rates)
+        self._dtw_requests = None
+        self._tables_raw = {f"raw_{k}": v for k, v in self._client.tables.items()}
+        self._tables = dict(sorted(IbSessionTws._make_tables(self._tables_raw).items()))
+
+    ####################################################################################################################
+    ####################################################################################################################
+    ## Connect / Disconnect / Subscribe
+    ####################################################################################################################
+    ####################################################################################################################
+
+    def connect(self, host: str = "", port: int = 7497, client_id: int = 0) -> None:
+        """Connect to an IB TWS session.  Raises an exception if already connected.
+
+        Args:
+            host (str): The host name or IP address of the machine where TWS is running. Leave blank to connect to the local host.
```

Comment:
Could combine this with __init__, so there is no chance for the user to call connect() more than one time, making _assert_connected unnecessary, ...

Reply:
I thought about that, but there are some complexities.  There are use cases where a program may want to connect, disconnect, and then reconnect again later.  If this is moved into init, that is not possible.  I have moved the parameters into the init method to make sure that if someone does this, they will only be getting data from the same source.

I think that there is a reasonable argument that we might want to remove `connect` and `disconnect` and just connect once in `__init__`.  The only problem is that if `disconnect` is not called, Python/IB holds the ports and won't allow other connection.